body_test.go

```diff
@@ -0,0 +1,189 @@
+package artifact
+
+import (
+	"bytes"
+	"crypto/rand"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+var allTheBytes = []byte{1, 3, 7, 15, 31, 63, 127, 255}
+
+func setup(t *testing.T) (*os.File, []byte, func()) {
+
+	var b bytes.Buffer
+	var err error
+
+	file, err := ioutil.TempFile("testdata", "body-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	out := io.MultiWriter(file, &b)
+
+	// We declare this outside of the loop to minimize allocations
+	randbuf := make([]byte, 8)
+
+	for i := 0; i < 256; i++ {
+		// Using repeated bytes is nice for debugging purposes
```

Comment:
Would that not make it harder, because you don't which of the repeated sections it was copied from? If they were random bytes, there is a good chance they could have only be copied from a single location from the original binary block, isn't there?

Reply:
it made it easier for me because I knew any number of bytes that was a multiple of 8 was going to have the same sequence.  Since I was trying to find off-by-one type things and using chunks which were sized as multiples of 8, I could spot the shift.  The bytes I chose even nicely printed into the whole screen and you could see a neat pattern.  It's probably served it's purpose and could be removed.... no strong feelings :)